examples/demo/app/main.go

```diff
@@ -23,81 +23,132 @@ import (
 	"os"
 	"time"
 
-	"contrib.go.opencensus.io/exporter/ocagent"
-	"go.opencensus.io/stats"
-	"go.opencensus.io/stats/view"
-	"go.opencensus.io/tag"
-	"go.opencensus.io/trace"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/baggage"
+	"go.opentelemetry.io/otel/exporters/otlp"
+	"go.opentelemetry.io/otel/label"
+	"go.opentelemetry.io/otel/metric"
+	"go.opentelemetry.io/otel/propagation"
+	"go.opentelemetry.io/otel/sdk/metric/controller/push"
+	"go.opentelemetry.io/otel/sdk/metric/processor/basic"
+	"go.opentelemetry.io/otel/sdk/metric/selector/simple"
+	"go.opentelemetry.io/otel/sdk/resource"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	"go.opentelemetry.io/otel/semconv"
+	"google.golang.org/grpc"
 )
 
-func main() {
-	ocAgentAddr, ok := os.LookupEnv("OTEL_AGENT_ENDPOINT")
+// Initializes an OTLP exporter, and configures the corresponding trace and
+// metric providers.
+func initProvider() func() {
+	ctx := context.Background()
+
+	otelAgentAddr, ok := os.LookupEnv("OTEL_AGENT_ENDPOINT")
 	if !ok {
-		ocAgentAddr = "0.0.0.0:55678"
+		otelAgentAddr = "0.0.0.0:55680"
 	}
-	oce, err := ocagent.NewExporter(
-		ocagent.WithAddress(ocAgentAddr),
-		ocagent.WithInsecure(),
-		ocagent.WithServiceName(fmt.Sprintf("example-go-%d", os.Getpid())))
-	if err != nil {
-		log.Fatalf("Failed to create ocagent-exporter: %v", err)
+
+	exp, err := otlp.NewExporter(
+		otlp.WithInsecure(),
+		otlp.WithAddress(otelAgentAddr),
+		otlp.WithGRPCDialOption(grpc.WithBlock()), // useful for testing
+	)
+	handleErr(err, "failed to create exporter")
+
+	res, err := resource.New(ctx,
+		resource.WithAttributes(
+			// the service name used to display traces in backends
+			semconv.ServiceNameKey.String("test-service"),
+		),
+	)
+	handleErr(err, "failed to create resource")
+
+	bsp := sdktrace.NewBatchSpanProcessor(exp)
+	tracerProvider := sdktrace.NewTracerProvider(
+		sdktrace.WithConfig(sdktrace.Config{DefaultSampler: sdktrace.AlwaysSample()}),
+		sdktrace.WithResource(res),
+		sdktrace.WithSpanProcessor(bsp),
+	)
+
+	pusher := push.New(
+		basic.New(
+			simple.NewWithExactDistribution(),
+			exp,
+		),
+		exp,
+		push.WithPeriod(7*time.Second),
+	)
+
+	// set global propagator to tracecontext (the default is no-op).
+	otel.SetTextMapPropagator(propagation.TraceContext{})
+	otel.SetTracerProvider(tracerProvider)
+	otel.SetMeterProvider(pusher.MeterProvider())
+	pusher.Start()
+
+	return func() {
+		handleErr(tracerProvider.Shutdown(ctx), "failed to shutdown provider")
+		handleErr(exp.Shutdown(ctx), "failed to stop exporter")
+		pusher.Stop() // pushes any last exports to the receiver
 	}
```

Comment:
@MrAlias can we have a helper in Otel-go that makes this a single call, to which you only pass the exporter and the resource attributes? The decisions that this code makes appear to be reasonable defaults that majority of users will want to make (I may be wrong). Unless we want to force them to make this decisions consciously a helper would be very useful.

Reply:
Yeah seems reasonable. We do that with other exporters (i.e. https://github.com/open-telemetry/opentelemetry-go/blob/fd3c82bec5fc8012da1fdd0402149b38bd8ec59d/exporters/stdout/exporter.go#L85-L93), looks like we haven't done it yet with OTLP.

Reply:
Tracking with https://github.com/open-telemetry/opentelemetry-go/issues/1347

Reply:
Thanks